ARK Server Manager/Lib/Model/ServerRuntime.cs

```diff
@@ -380,23 +393,117 @@ public static void SendStop(Process proc)
                 // Run the SteamCMD to install the server
                 var steamCmdPath = Updater.GetSteamCMDPath();
                 //DataReceivedEventHandler dataReceived = (s, e) => Console.WriteLine(e.Data);
-                var success = await ServerUpdater.UpgradeServerAsync(validate, this.ProfileSnapshot.InstallDirectory, steamCmdPath, Config.Default.SteamCmdInstallServerArgsFormat, null /* dataReceived*/, cancellationToken);
-                if (success && ServerManager.Instance.AvailableVersion != null)
+                updateResult.ServerUpdated = await ServerUpdater.UpgradeServerAsync(validate, this.ProfileSnapshot.InstallDirectory, steamCmdPath, Config.Default.SteamCmdInstallServerArgsFormat, null /* dataReceived*/, cancellationToken);
+                if (updateResult.ServerUpdated && ServerManager.Instance.AvailableVersion != null)
                 {
                     this.Version = ServerManager.Instance.AvailableVersion;
                 }
 
-                return success;
+                // Run the SteamCMD to install the mods
+                if (updateMods)
+                {
+                    if (updateResult.ServerUpdated)
+                    {
+                        updateResult.ModsUpdated = await ModUpdater.UpgradeAsync(this.ProfileSnapshot.InstallDirectory, steamCmdPath, Config.Default.SteamCmdInstallModArgsFormat, this.ProfileSnapshot.ModIdString, this.ProfileSnapshot.MapIdString, null /* dataReceived*/, cancellationToken);
+                    }
+                }
+                else
+                    updateResult.ModsUpdated = true;
+
+                return updateResult;
             }
             catch (TaskCanceledException)
             {
-                return false;
+                return new UpdateResult { ServerUpdated = false, ModsUpdated = false };
             }
             finally
             {
                 this.Status = ServerStatus.Stopped;
             }
-        }       
+        }
+
+        public async Task<UpdateResult> UpgradeModsAsync(CancellationToken cancellationToken)
+        {
+            var updateResult = new UpdateResult { ServerUpdated = false, ModsUpdated = false };
+
+            if (!System.Environment.Is64BitOperatingSystem)
+            {
+                var result = MessageBox.Show("ARK: Survival Evolved(tm) Server requires a 64-bit operating system to run.  Your operating system is 32-bit and therefore the Ark Server Manager will be unable to start the server, but you may still install it or load and save profiles and settings files for use on other machines.\r\n\r\nDo you wish to continue?", "64-bit OS Required", MessageBoxButton.YesNo, MessageBoxImage.Warning);
```

Comment:
This needs to come from the localization file.  Were you able to get the run-time query of loc strings from XAML to work?